torch/distributed/distributed_c10d.py

```diff
@@ -929,11 +935,32 @@ def all_reduce(tensor,
         Async work handle, if async_op is set to True.
         None, if not async_op or if not part of the group
 
+    Example:
+        Tensors are all of dtype torch.int64.
+        We have 2 process groups, 2 ranks.
+        rank 0 passes:
+            tensor = [[1, 1], [2, 2]]
+        rank 1 passes:
+            tensor = [[3, 3], [4, 4]]
+        both rank 0 and 1 get:
+            tensor = [[4, 4], [6, 6]]
+
+        Tensors are all of dtype torch.complex64.
+        We have 2 process groups, 2 ranks.
+        rank 0 passes:
+            tensor = [[1+i, 1+i], [2+2i, 2+2i]]
+        rank 1 passes:
+            tensor = [[3+3i, 3+3i], [4+4i, 4+4i]]
+        both rank 0 and 1 get:
+            tensor = [[4+4i, 4+4i], [6+6i, 6+6i]]
```

Comment:
could you please build the doc to verify that this renders correctly?

Reply:
synced privately on this- I updated the doc to use python repl formatting for the examples and tested out building the doc:

![image](https://user-images.githubusercontent.com/16311747/95377462-9c087b80-08b0-11eb-88cf-09fa8dec64cd.png)